### PR TITLE
Incident casebook link route

### PR DESCRIPTION
**Pull request by gbuisson**
epic https://github.com/threatgrid/iroh/issues/1940

closes  https://github.com/threatgrid/iroh/issues/1964

Also, moved bulk and bundle test ns to their correct location.

### CODE REVIEW

File: test/ctia/link/routes_test.clj [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+         (is (= 201 incident-status))
+         (is (= 201 link-status))
+
+         (is (= (:id casebook-body)
+                (:source_ref link-response))
+             "The New Relationship targets the incident")
+
+         (is (= (:id incident-body)
+                (:target_ref link-response))
+             "The New Relationship sources the casebook"))))))
```
Comment (msprunck) [inline test/ctia/link/routes_test.clj]: Could you check that the relationship exists in the store? I'm not sure that the result comes from the store.

File: src/ctia/link/routes.clj [hunk 2]
```diff
@@ -0,0 +1,75 @@
+(ns ctia.link.routes
```
Comment (msprunck) [inline src/ctia/link/routes.clj]: Why not add this route to the `ctia.entity.incident` or `ctia.entity.relationship` ns? For me a link between a casebook and a incident is a `Relationship`.

**Comment by gbuisson** [comment]
I think I addressed all feedback

File: src/ctia/entity/relationship.clj [hunk 3]
```diff
@@ -62,6 +71,69 @@
...
+
+(def incident-casebook-link-route
+  (POST "/:id/link" []
+        :return rs/Relationship
+        :body [link-req IncidentCasebookLinkRequest
+               {:description "an Incident Link request"}]
+        :header-params [{Authorization :- (s/maybe s/Str) nil}]
+        :summary "Link an Incident to a Casebook"
+        :path-params [id :- s/Str]
+        :capabilities #{:create-incident
```
Comment (msprunck) [inline src/ctia/entity/relationship.clj]: Why the `:create-incident` capability is required? Only a Relationship is created.

File: src/ctia/entity/relationship.clj [hunk 4]
```diff
@@ -62,6 +71,69 @@
...
+                                       long-id->id
+                                       :short-id)
+                                   identity-map
+                                   {})]
+
+          (cond
+            (not incident)
+            (not-found)
+            (not casebook)
+            (not-found)
```
Comment (msprunck) [inline src/ctia/entity/relationship.clj]: IMO, if the casebook is not found, it's a bad request. The `not-found` response is only related to the request URI.

**Comment by hshahman** [comment]
Verified in rel-v1.4

**Pull request merged**